### PR TITLE
Bump to newest nightly

### LINKED ISSUE #114: Build fails with recent nightly
It looks like a recent nightly has renamed the internal `syntax` crate to `rustc_ast`, and the `syntax_pos` crate seems gone as well, though I've yet to find where it went.

Edit: Ah, `syntax_pos` is now `rustc_span`

Edit 2: Doesn't seem like a simple find/replace, other stuff has changed in rustc as well :disappointed: It might be nice to pin the toolchain to the last nightly that can be used to build.

### PR DESCRIPTION
**Pull request by Xanewok**
Closes #114

Opening as a draft, because I fixed all of the compilation errors but I need to double-check the changes made and fix the tests (which are now failing)

cc @gnunicorn

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 // Needs a nightly feature, doesn't bring that much to the table
 // FIXME: Apply Clippy lint once or-patterns are stabilized (rust-lang/rust#54883)
 #![allow(clippy::unnested_or_patterns)]
-#![deny(warnings)]
+// #![deny(warnings)]
```
Comment (gnunicorn) [inline src/lib.rs:8]: I guess that should not be merged as is.

File: src/translate.rs [hunk 2]
```diff
@@ -364,107 +364,108 @@ impl<'a, 'tcx> TranslationContext<'a, 'tcx> {
         index_map: &HashMap<u32, DefId>,
         predicate: Predicate<'tcx>,
     ) -> Option<Predicate<'tcx>> {
+        use rustc_middle::ty::PredicateAtom;
```
Comment (gnunicorn) [inline src/translate.rs:367]: I guess this should eventually be moved in the `use` below...

Comment (Xanewok) [reply src/translate.rs:367]: Yep, that's draft for now. I need to be extra careful around the predicates still, so I'll move the use statement around once I revisit the commit that introduced that and fix remaining tests.

**Comment by Xanewok** [comment]
This should be now ready to review.

Each commit bumping nightly version should link to the breaking change upstream (except the last one) to facilitate reviewing. The only non-trivial changes are https://github.com/rust-lang/rust-semverver/pull/144/commits/0d52c24b9fd72dd48ed4d81ad45b377d18f1fb44 and https://github.com/rust-lang/rust-semverver/pull/144/commits/6604643bfb30a1c68f66ca908f0591648ec5380f.

r? @Manishearth or @nikomatsakis (since you helped review https://github.com/rust-lang/rust-semverver/pull/144/commits/0d52c24b9fd72dd48ed4d81ad45b377d18f1fb44)

**Pull request merged**